lib/action_view/helpers/dynamic_form.rb

```diff
@@ -262,29 +264,55 @@ def default_input_block
         Proc.new { |record, column| %(<p><label for="#{record}_#{column.name}">#{column.human_name}</label><br />#{input(record, column.name)}</p>) }
       end
 
-      module InstanceTagMethods
-        def to_tag(options = {})
+
+      module InputBuilderMethods
+        DEFAULT_MAXLENGTH = 30
+
+        def initialize(object_name, method, context, options)
+          @object_name = object_name
+          @context = context
+          @method = method
+          @options = options
+        end
+
+        def to_tag
           case column_type
-            when :string
-              field_type = @method_name.include?("password") ? "password" : "text"
-              to_input_field_tag(field_type, options)
-            when :text
-              to_text_area_tag(options)
-            when :integer, :float, :decimal
-              to_input_field_tag("text", options)
-            when :date
-              to_date_select_tag(options)
-            when :datetime, :timestamp
-              to_datetime_select_tag(options)
-            when :time
-              to_time_select_tag(options)
-            when :boolean
-              to_boolean_select_tag(options).html_safe
+          when :string
+            @options["type"] = @method.include?("password") ? "password" : "text"
+            Tags::TextField.new(*generic_args).render
+          when :text
+            Tags::TextArea.new(*generic_args).render
+          when :integer, :float, :decimal
+            Tags::TextField.new(*generic_args).render
+          when :date
+            Tags::DateField.new(*generic_args).render
+          when :datetime, :timestamp
+            Tags::DatetimeLocalField.new(*generic_args).render
+          when :time
+            Tags::TimeField.new(*generic_args).render
+          when :boolean
+            Tags::CheckBox.new(@object_name, @method, @context, "1", "0", @options).render
```

Comment:
It makes sense for this tag to be a checkbox input instead of a select, as previously https://api.rubyonrails.org/v3.1.3/classes/ActionView/Helpers/InstanceTag.html#method-i-to_boolean_select_tag